scotty/density_fit.py

```diff
@@ -35,10 +35,12 @@ def __init__(self, poloidal_flux_enter: float):
 
     def __call__(self, poloidal_flux: ArrayLike) -> ArrayLike:
         """Returns the interpolated density at ``poloidal_flux`` points."""
+        tol = 1e-10
+
         poloidal_flux = np.asfarray(poloidal_flux)
         density = np.asfarray(self._fit_impl(poloidal_flux))
         # Mask density inside plasma
-        is_inside = poloidal_flux <= self.poloidal_flux_enter
+        is_inside = poloidal_flux <= self.poloidal_flux_enter + tol
```

Comment:
I think we should back out this change. Because most of the fits are defined to go to zero exactly at `poloidal_flux_enter`, this gives rise to a negative density when `poloidal_flux_enter < poloidal_flux < poloidal_flux_enter + tol`

Reply:
Perhaps in the discontinuous case, we can have electron_density_plasma to be evaluated at poloidal_flux_enter-tol